felix/bpf/ut/precompilation_test.go

```diff
@@ -41,66 +41,70 @@ func TestPrecompiledBinariesAreLoadable(t *testing.T) {
 		logLevel := logLevel
 		// Compile the TC endpoint programs.
 		logCxt := log.WithField("logLevel", logLevel)
-		for _, epToHostDrop := range []bool{false, true} {
-			epToHostDrop := epToHostDrop
-			logCxt = logCxt.WithField("epToHostDrop", epToHostDrop)
-			for _, fibEnabled := range []bool{false, true} {
-				fibEnabled := fibEnabled
-				logCxt = logCxt.WithField("fibEnabled", fibEnabled)
-				epTypes := []tc.EndpointType{
-					tc.EpTypeWorkload,
-					tc.EpTypeHost,
-					tc.EpTypeTunnel,
-					tc.EpTypeWireguard,
-				}
-				for _, epType := range epTypes {
-					epType := epType
-					logCxt = logCxt.WithField("epType", epType)
-					if epToHostDrop && epType != tc.EpTypeWorkload {
-						log.Debug("Skipping combination since epToHostDrop only affect workloads")
-						continue
+		for _, btfEnabled := range []bool{false, true} {
```

Comment:
would this work on a kernel that is not btf enabled? I think we want to run just the legacy non-co-re on such kernels and both on co-re capable kernels, right?